Adding optional Sentinels

pcejrowski opened the pull request:
This PR adds 3 optional Sentinels on ports `5000`(name: `sentinel7000`), `5001` (`sentinel7001`), and `5002` (`sentinel7001`), that map to master nodes running on `7000`, `7001`, and `7002`, respectively.
They are not running by default, but have to be turned on using `-e "SENTINEL=true"` flag.
I have temporarily pushed the image to Docker Hub under `pcej/redis-cluster:latest` if anyone wants to test.
I have been running the Docker container with:
```
docker run \
  -e "SENTINEL=true" \
  -e "IP=0.0.0.0" \
  -p 7000:7000 \
  -p 5000:5000 \
  --name redis-cluster \
  pcej/redis-cluster:latest
```
and then testing with:
```
redis-cli -p 5000
127.0.0.1:5000> sentinel master sentinel7000
```

pcejrowski commented:
Hi @Grokzen, do you have any opinion about this PR? Can I expect to have it merged soon, or should I maintain my fork?

Grokzen commented:
@pcejrowski I have been conflicted myself with the decision about having standalone redis instances inside the current container as that could be provided elsewhere in some more standrad/basic container. This one falls in the same category really. I see a great benefit of having all 3 different types of cluster/non-cluster solutions inside one container, but at the same time i am probably leaning more towards splitting the containers into 3 different containers where one is a standalone instance, one is sentinel and one is cluster. Then you have the option to choose freely based on your own needs.

But i might merge this and rework towards that idea later on.

Grokzen commented:
@pcejrowski I would like the following changes to be added if possible before i can merge this. One bugfix in docker-compose file, some readme updates and enable this to be configured when using docker-compose.

```
diff --git a/README.md b/README.md
index 7475976..92d3aef 100644
--- a/README.md
+++ b/README.md
@@ -85,6 +85,15 @@ Set env variable CLUSTER_ONLY=true.
       docker run ... -e CLUSTER_ONLY=true ...
 
 
+## Include sentinel instances
+
+Sentinel instances is not enabled by default.
+
+When running with docker-compose set the environment variable on your system `REDIS_USE_SENTINEL=true` and start your container.
+
+If running with plain docker send in `-e SENTINEL=true`.
+
+
 ## Build alternative redis versions
 
 For a release to be buildable it needs to be present at this url: http://download.redis.io/releases/
diff --git a/docker-compose.yml b/docker-compose.yml
index 68c3890..8ebd651 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -3,6 +3,7 @@ services:
   redis-cluster:
     environment:
      IP: ${REDIS_CLUSTER_IP}
+     SENTINEL: ${REDIS_USE_SENTINEL}
     build:
       context: .
       args:
@@ -10,4 +11,4 @@ services:
     hostname: server
     ports:
       - '7000-7007:7000-7007'
-      - '5000-5002:5000:5002'
+      - '5000-5002:5000-5002'
```

pcejrowski commented:
Added those 3 changes and renamed the commit message to follow the repository conventions. 

BTW, I had enabled  `Allow edits from maintainers.`, so you were actually able to push those changes to my branch.

Grokzen commented:
Ouch O.o i would not had enabled that feature tho :P Usually i do not want to mess with anyones fork tbh

Grokzen commented:
@pcejrowski LGTM, tested locally and looks fine. I will hold off pushing new images as i have some other changes for the standalone instances in the works.
